src/app/pages/Register/Register.tsx

```diff
@@ -1,20 +1,30 @@
 import React, { useState } from 'react';
+import { useHistory } from 'react-router-dom';
 import LabeledInput from '../../components/LabeledInput/LabeledInput';
 import styles from './Register.module.css';
 import Button from '../../components/Button/Button';
 import BackButton from '../../components/BackButton/BackButton';
 import ProfilePictureIcon from '../../components/Icons/ProfilePictureIcon';
 
 function RegisterForm(): JSX.Element {
+  const history = useHistory();
   const [firstName, setFirstName] = useState('');
   const [lastName, setLastName] = useState('');
   const [email, setEmail] = useState('');
   const [password, setPassword] = useState('');
   const [confirmPassword, setConfirmPassword] = useState('');
 
-  function handleSubmit(event: React.FormEvent<HTMLFormElement>) {
+  async function handleSubmit(event: React.FormEvent<HTMLFormElement>) {
     event.preventDefault();
-    alert(`${firstName} submitted`);
+    const user = { email, firstName, lastName, password };
```

Comment:
Good job. I would feel more safe if you add `User` type here:
https://github.com/lmachens/photoplay/blob/main/src/types.ts#L27

```suggestion
    const user: User = { email, firstName, lastName, password };
```